src/backend/marsha/core/api.py

```diff
@@ -678,6 +686,110 @@ def initiate_upload(self, request, pk=None):
         return Response(response)
 
 
+class LiveRegistrationViewSet(
+    ObjectPkMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Viewset for the API of the LiveRegistration object."""
+
+    permission_classes = [permissions.IsVideoToken]
+    queryset = LiveRegistration.objects.all()
+    serializer_class = serializers.LiveRegistrationSerializer
+
+    def get_queryset(self):
+        """Restrict list access to liveRegistration related to the video in the JWT token."""
+        user = self.request.user
+        if isinstance(user, TokenUser) and user.token.payload is not None:
+            # admin and instructors can access all registrations
+            if user.token.payload.get("roles", None) is not None and any(
+                role in ["administrator", "instructor"]
+                for role in user.token.payload["roles"]
+            ):
+                return LiveRegistration.objects.filter(
+                    id=self.kwargs["pk"], video__id=user.id
+                )
+            # others can only read their registrations
+            if user.token.payload.get("user", None) is not None:
+                if user.token.payload["user"].get("email", None) is not None:
+                    # token has an email
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        video__id=user.id,
+                        email=user.token.payload["user"]["email"],
+                    )
+                # no email, user has access to this registration if it's the right combinaison of
+                # lti_user_id and lti_consumer_id
+                if (
+                    user.token.payload["user"].get("id", None) is not None
+                    and user.token.payload.get("context_id", None) is not None
+                ):
+                    playlist = Playlist.objects.get(
+                        lti_id=user.token.payload["context_id"]
+                    )
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        lti_consumer_site=playlist.consumer_site,
+                        lti_user_id=user.token.payload["user"]["id"],
+                        video__id=user.id,
+                    )
+
+        return LiveRegistration.objects.none()
+
+    def perform_create(self, serializer):
```

Comment:
I think this should be done in the serailizer itself and not here. the `create` method in the serializer is made for this.

Reply:
I actually wondered about it and was pretty sure we would speak about it :-). Reading the documentation I didn't find that clear. I wonder what's the perform_create for and when to use this one over the create method? Do you have a better understanding than me on the difference? I ended up making this choice thinking as it was before the serializer, it was better to do it here so the serializer could add up its regular controls.

Reply:
I've moved everything to the serializer.